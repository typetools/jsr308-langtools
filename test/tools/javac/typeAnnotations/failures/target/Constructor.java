/*
 * @test /nodynamiccopyright/
 * @bug 6843077
 * @summary test invalid location of TypeUse
 * @author Mahmood Ali
 * @compile/fail/ref=Constructor.out -XDrawDiagnostics -source 1.8 Constructor.java
 */

import java.lang.annotation.Target;
import java.lang.annotation.ElementType;

class Constructor {
  // Declaration annotation, not type use annotation
  @A Constructor() { }

  // Declaration annotation, not type parameter annotation
  @B Constructor(int x) { }

  // Allowed!
  Constructor(@A Constructor this, Object o) { }

  // Forbidden location for type parameter annotation
  Constructor(@B Constructor this, Object o1, Object o2) { }
}

class Constructor2 {
  class Inner {
    // OK
    Inner(@A Inner this) { }
  }
}

@Target(ElementType.TYPE_USE)
@interface A { }

@Target(ElementType.TYPE_PARAMETER)
@interface B { }

