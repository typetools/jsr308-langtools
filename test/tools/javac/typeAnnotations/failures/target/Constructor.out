Constructor.java:14:3: compiler.err.annotation.type.not.applicable
Constructor.java:17:3: compiler.err.annotation.type.not.applicable
Constructor.java:23:15: compiler.err.annotation.type.not.applicable
3 errors